### ISSUE TITLE
Support for v5.0.0

### ISSUE THREAD

**Issue by mickeyvip** [Reactive-Extensions/RxJS-DOM#105]
Hello.

The version 5.0.0 of `Rx` (https://github.com/ReactiveX/RxJS) is in beta.

Will `RxJS-DOM` be supporting the new version?

Thank you.

**Comment by mattpodwysocki** [comment]
@mickeyvip yes, but it will be built in at some point. cc @trxcllnt who is working on just that, although some things need to be addressed such as how `fromEvent` differs here than in the `fromEvent` in RxJS core.

**Comment by trxcllnt** [comment]
@mickeyvip @mattpodwysocki so far [ReactiveX/RxJS](https://github.com/ReactiveX/RxJS/) has [`Observable.ajax`, `Observable.ajax.get`, `Observable.ajax.put`, `Observable.ajax.post`, `Observable.ajax.delete`, `Observable.ajax.getJSON`](https://github.com/ReactiveX/RxJS/blob/master/src/observable/dom/ajax.ts#L41), [`Observable.webSocket`](https://github.com/ReactiveX/RxJS/blob/master/src/observable/dom/webSocket.ts), and [`Scheduler.animationFrame`](https://github.com/ReactiveX/RxJS/blob/master/src/Rx.DOM.ts#L133).

You should be able to access these via `import {Observable, Scheduler} from rxjs/Rx.DOM;`

**Comment by mattpodwysocki** [comment]
@trxcllnt are we planning on bringing all of the DOM stuff over relatively soon?

**Comment by trxcllnt** [comment]
@mattpodwysocki I don't see issues for the things we don't have yet, but we should file them. I forgot we also already have the microtask scheduler (aka the [asapScheduler](https://github.com/ReactiveX/RxJS/blob/master/src/scheduler/AsapScheduler.ts)) implemented.

From what I can tell, we're missing the following items:

- [events](https://github.com/Reactive-Extensions/RxJS-DOM/blob/master/src/events/events.js)
- [ready](https://github.com/Reactive-Extensions/RxJS-DOM/blob/master/src/events/ready.js)
- [jsonp](https://github.com/Reactive-Extensions/RxJS-DOM/blob/master/src/ajax/jsonp.js)
- [fromEventSource](https://github.com/Reactive-Extensions/RxJS-DOM/blob/master/src/dom/eventsource.js)
- [fromReader](https://github.com/Reactive-Extensions/RxJS-DOM/blob/master/src/dom/fromreader.js)
- [geolocation](https://github.com/Reactive-Extensions/RxJS-DOM/blob/master/src/dom/geolocation.js)
- [fromMutationObserver](https://github.com/Reactive-Extensions/RxJS-DOM/blob/master/src/dom/mutationobserver.js)
- [worker](https://github.com/Reactive-Extensions/RxJS-DOM/blob/master/src/dom/worker.js)

**Comment by mickeyvip** [comment]
@mattpodwysocki, @trxcllnt,  thank you so much!

**Comment by franciclo** [comment]
Is there any way to use it with rxjs 5 now?
is kind of tedious to dispose some streams as .dispose() and other as .unsubscribe()

**Comment by trxcllnt** [comment]
@franciclo If you have code written with Rx4 naming conventions, the quickest solution is to alias from new to old up front. Here's what we're currently aliasing in our projects at work:

``` es6
import Rx from 'rxjs';

Rx.Observable.return = (value) => Rx.Observable.of(value);

Rx.Subject.prototype.onNext = Rx.Subject.prototype.next;
Rx.Subject.prototype.onError = Rx.Subject.prototype.error;
Rx.Subject.prototype.onCompleted = Rx.Subject.prototype.complete;
Rx.Subject.prototype.dispose = Rx.Subscriber.prototype.unsubscribe;
Rx.AsyncSubject.prototype.onNext = Rx.AsyncSubject.prototype.next;
Rx.AsyncSubject.prototype.onCompleted = Rx.AsyncSubject.prototype.complete;
Rx.BehaviorSubject.prototype.onNext = Rx.BehaviorSubject.prototype.next;
Rx.ReplaySubject.prototype.onNext = Rx.ReplaySubject.prototype.next;

Rx.Subscriber.prototype.onNext = Rx.Subscriber.prototype.next;
Rx.Subscriber.prototype.onError = Rx.Subscriber.prototype.error;
Rx.Subscriber.prototype.onCompleted = Rx.Subscriber.prototype.complete;
Rx.Subscriber.prototype.dispose = Rx.Subscriber.prototype.unsubscribe;

Rx.Subscription.prototype.dispose = Rx.Subscription.prototype.unsubscribe;
```

**Comment by franciclo** [comment]
Right, that will patch it up, thank you for your time
